recovery/util.cpp

```diff
@@ -39,6 +39,17 @@ void putFileContents(const QString &filename, const QByteArray &data)
     f.close();
 }
 
+bool backup(QString filename, QString ext)
+{
+
+    QString backupName = filename + "." + ext;
+    const char * backupFile = backupName.toUtf8().constData();
```

Comment:
Just out of curiosity, why the switch away from the Qt functions (QString, QFile::remove, QFile::rename, etc.) ? I think that _might_ allow you to get rid of the ugly `.toUtf8().constData()` ?

Reply:
Sorry, I took your pseudo code too literally :) - I will revert.